### ISSUE TITLE
StyleGAN-NADA: CLIP-Guided Domain Adaptation of Image Generators

### ISSUE THREAD

**Issue by nmhkahn** [nmhkahn/paper-notes#5]
## TL;DR
- StyleCLIP과 동일하게 text prompt를 사용해서 이미지 manipulation 하자
- 그러나 StyleCLIP은 pretrained StyleGAN을 사용해서 in-domain 데이터에 대해서만 생성/변환이 가능하고, out-of-domain에 대해서는 처리를 못한다는 한계가 있음
- 이를 해결하기 위해 directional CLIP loss와 layer-freezing 기법 도입

## CLIP loss
기존 (global) CLIP loss는
- Diversity 관점에서 인센티브가 없기 때문에 mode-collapse가 일어날 우려가 큼
- CLIP 모델을 그냥 fool 해버려서 clip score는 매우 좋은데 생성 이미지는 형편없어지는 경우가 많이 생김

그래서 직접적으로 text와 image의 CLIP score를 줄이지 말고 direction을 줄이자

![](https://user-images.githubusercontent.com/16869368/128661119-286c0d29-f29c-4ef9-b30f-dce63025b663.png)
![](https://user-images.githubusercontent.com/16869368/128661064-303b3e1e-7a13-40e0-8b63-22efd5b03999.png)

이미지에 생성되는 아티팩트가 CLIP 임베딩의 norm에 연관이 있어서 이것도 loss에 추가함 (StyleCLIP도 비슷한 loss가 있긴 함)
![](https://user-images.githubusercontent.com/16869368/128661376-a2146d4b-fa9b-49b2-bc32-941a9a1d5d90.png)

## Layer-Freezing
![](https://user-images.githubusercontent.com/16869368/128661436-c63458aa-746b-4f5c-871f-92faa67ffa25.png)
- 기존 few-shot 이미지 생성 모델은 subnetwork의 파라미터를 어느정도 제한하는게 좋다고 함
- 이미지에 대한 일반적인 역할을 하는 파라미터는 source, target 모두 도움되기도 하고, 필요한것만 업데이트 해야 overfit에서 자유로움
- 여기서는 일단 global CLIP loss로 레이어를 업데이트 해보고, significant 하게 변화하는 k개 고른다음에 이거만 optimization 하는 방식 사용함
